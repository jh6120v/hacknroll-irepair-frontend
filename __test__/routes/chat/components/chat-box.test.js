import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import ChatBox from '../../../../src/routes/chat/components/chat-box';

describe('test chat-box component.', () => {
    it('test <ChatBox /> snapshot.', () => {
        cleanup();
        const { container } = render(<ChatBox />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ChatBox /> snapshot render with isSelf equal to true', () => {
        cleanup();
        const { container } = render(<ChatBox isSelf />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ChatBox /> snapshot render with images props.', () => {
        cleanup();
        const { container } = render(<ChatBox images="images" />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <ChatBox /> when render different props.', () => {
        cleanup();

        const { getByTestId } = render(<ChatBox author="AUTHOR" message="MESSAGE" time="TIME" />);

        expect(getByTestId('author').innerHTML).toEqual('AUTHOR');
        expect(getByTestId('message').innerHTML).toEqual('MESSAGE');
        expect(getByTestId('time').innerHTML).toEqual('TIME');
    });
});
