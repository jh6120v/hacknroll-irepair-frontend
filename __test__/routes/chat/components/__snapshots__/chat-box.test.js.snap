// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test chat-box component. test <ChatBox /> snapshot render with images props. 1`] = `
<div
  class="sc-bxivhb ksdOAy"
>
  <div
    class="sc-ifAKCX kWalcW"
  >
    <div
      class="sc-bZQynM dZOkkL"
    />
    <div
      class="sc-gzVnrw kwznsF"
    >
      <div
        class="sc-htoDjs nQyXR"
        data-testid="author"
      />
      <div
        class="sc-iwsKbI fLEBAd"
      >
        <img
          src="images"
        />
      </div>
    </div>
  </div>
  <div
    class="sc-EHOje iFzWVO"
    data-testid="time"
  />
</div>
`;

exports[`test chat-box component. test <ChatBox /> snapshot render with isSelf equal to true 1`] = `
<div
  class="sc-bxivhb hjdqxo"
>
  <div
    class="sc-ifAKCX dttElF"
  >
    <div
      class="sc-gzVnrw LkRjp"
    >
      <div
        class="sc-dnqmqq bzTNDS"
        data-testid="message"
      />
    </div>
  </div>
  <div
    class="sc-EHOje iFzWVO"
    data-testid="time"
  />
</div>
`;

exports[`test chat-box component. test <ChatBox /> snapshot. 1`] = `
<div
  class="sc-bxivhb ksdOAy"
>
  <div
    class="sc-ifAKCX kWalcW"
  >
    <div
      class="sc-bZQynM dZOkkL"
    />
    <div
      class="sc-gzVnrw kwznsF"
    >
      <div
        class="sc-htoDjs nQyXR"
        data-testid="author"
      />
      <div
        class="sc-dnqmqq dKlddl"
        data-testid="message"
      />
    </div>
  </div>
  <div
    class="sc-EHOje iFzWVO"
    data-testid="time"
  />
</div>
`;
